File: src/AprioritWebCalendar/ClientApps/CalendarModule/src/app/calendar/models/calendar.ts
import { User } from "../../authentication/models/user";
import { EventCalendar } from "./event.calendar";

export class Calendar {
    public Id: Number;
    public Name: string;
    public Description: string;
    public Color: string;
    public Owner: User;

    public IsReadOnly?: boolean;
    public IsSubscribed?: boolean;
    public IsDefault: boolean;

    public Events: EventCalendar[];
}

File: src/AprioritWebCalendar/AprioritWebCalendar.Business/DomainModels/Calendar.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AprioritWebCalendar.Business.DomainModels
{
    public class Calendar
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Color { get; set; }
        public User Owner { get; set; }

        public bool? IsDefault { get; set; }

        public IEnumerable<UserCalendar> SharedUsers { get; set; } = new List<UserCalendar>();
        public IEnumerable<EventCalendar> Events { get; set; } = new List<EventCalendar>();
    }
}
